lib/screens/util.ex

```diff
@@ -179,4 +179,16 @@ defmodule Screens.Util do
   def to_set(id) when is_binary(id), do: MapSet.new([id])
   def to_set(ids) when is_list(ids), do: MapSet.new(ids)
   def to_set(%MapSet{} = already_a_set), do: already_a_set
+
+  @doc """
+    Calculates the service day after the given DateTime (which must be in UTC).
+    For context, MBTA service days end at 3am, not at midnight.
+    So getting the next service day means adding 21 hours to current date, not 24.
+    To avoid duplicate DateTime calculations existing throughout the code,
+    this function will handle the actual calculations needed to get the next service day.
+  """
+  @spec get_service_day_tomorrow(DateTime.t()) :: DateTime.t()
+  def get_service_day_tomorrow(now) do
+    DateTime.add(now, 60 * 60 * 21)
```

Comment:
Sorry to keep hounding you on this but I'm worrying about a few things—date/time manipulation is a frequent source of bugs.

- Should it just return a `Date` struct, no time?
- Does this still work outside of DST when the time difference between Eastern and UTC is different? We're currently in EDT, UTC-4, but during the winter it's EST, UTC-5.
- Would guidance around how to use the return value help?
- Unit tests?

This is tricky because we mostly work with UTC datetimes, but the logic here is specifically relative to 3am _Eastern_ time.

How would you feel about making the code more explicit about what it's doing? Like:
```ex
now_eastern = DateTime.shift_zone(now, "America/New_York")
if now_eastern.hour >= 3 do
  # Confirmed that this just grabs the year/month/day fields without regard to timezone
  DateTime.to_date(now_eastern)
else
  Date.add(now_eastern, -1)
end
```

This way, the function would return a Date that has no sense of timezone and simply represents a service date that we can pass in our API queries.

Reply:
No need to apologize! Dates are super tricky and you brought up some good points I didn't think about.

Your logic made sense, but just needed a minor tweak. If we are between 12am and 3am, we are currently in the day of the next service day while still being in the previous service day (😵). That means we return the current day to get the next service day. All other times, we can safely add one day to get the correct date.

I added a couple of tests to help instill confidence. I made sure to first create a date in EST and then convert that date to UTC.